### DISCUSSION TITLE
How difficult was it for you to update the expo SDK version? What was it like to carry out a successful update?

### DISCUSSION THREAD

**Discussion by ildaneta** [expo/expo#24492] [Proposals and Ideas]
How difficult was it for you to update the expo SDK version? What was it like to carry out a successful update?

I would like to know your opinion about the expo SDK update process, as for me it was not such a memorable experience. Because of this, I created this feature suggestion for Expo, which would be to create a diff page between the SDK files of different versions, for example a diff between the SDK 47 and SDK 49 versions.

I would like your opinion.

My suggestion:
https://expo.canny.io/feature-requests/p/improvements-and-suggestions-for-expo-version-updates-creating-a-diff-page-like

Regards, Ilda

**Comment by tsapeta** [comment]
What issues have you encountered during the upgrade? As far as I can see, there is nothing about it here and on canny.
People generally don't complain about updates, especially if you use development builds. If you don't, it's all about applying changes necessary for React Native, not Expo where you usually just need to upgrade the `expo` package and then run `npx expo install --fix` to upgrade other SDK packages.

**Reply by ildaneta** [reply]
I found many differences, and the differences were not resolved by running `npx expo install --fix`. @tsapeta 

The way it was explained, it seems to be simple to update the versions, which in my opinion should be very simple, but that is not the case.

I can cite some examples but keeping in mind that this situation applies to all files, where, in the react native update documentation it says to modify some parts, however, due to the project having expo, there are specific parts of the expo that are not covered in the react native documentation and are not resolved with `npx expo install --fix`

Example 1 Podfile file
Already in the first line you can notice the difference

React Native documentation
![Screenshot 2023-09-19 at 10 10 21](https://github.com/expo/expo/assets/21963291/eaf47599-49a7-46a4-9f6e-d3595c3cecc9)

What really needed to be done
![Screenshot 2023-09-19 at 10 11 18](https://github.com/expo/expo/assets/21963291/65044e7c-2de4-4ae5-93a6-d624cf8b167d)

Add:
```
ENV['RCT_NEW_ARCH_ENABLED'] = podfile_properties['newArchEnabled'] == 'true' ? '1' : '0'
ENV['EX_DEV_CLIENT_NETWORK_INSPECTOR'] = podfile_properties['EX_DEV_CLIENT_NETWORK_INSPECTOR']
``` 

And explicitly let the target iOS:
`platform :ios, podfile_properties['ios.deploymentTarget'] || '13.0'`

If you perform a diff of all the files mentioned in the react native documentation in comparison with these same files from an expo project in the SDK 49 version, it is possible to notice that they are very different and this greatly hinders the update process.


Example 2 `AppDelegate.h`

React native documentation
![Screenshot 2023-09-19 at 10 37 25](https://github.com/expo/expo/assets/21963291/9142dc2d-18c1-4d46-9b25-747458d3cac7)

Project with expo

It's necessary remove `#import <Foundation/Foundation.h>`
And instead of `@interface AppDelegate : RCTAppDelegate` is necessary edit to `@interface AppDelegate : EXAppDelegateWrapper`

![Screenshot 2023-09-19 at 10 39 06](https://github.com/expo/expo/assets/21963291/1df51314-8bfc-48a4-a479-e11ea0f14dfb)

**Comment by jamespageced** [comment]
diff example is to much, so...

![image](https://github.com/user-attachments/assets/94d7fa3a-27bc-4ba9-85ab-4cec07423193)
